docs/src/spaces.md

```diff
@@ -0,0 +1,155 @@
+# [**9.** Scratchspaces](@id Scratchspaces)
+
+!!! compat "Julia 1.6"
+    Pkg's Scratchspaces functionality requires at least Julia 1.6.
+
+`Pkg` can manage and lifecycle scratchspaces of temporary or readily-recreatable data.
+These spaces can contain datasets, text, binaries, or any other kind of data that would be convenient to store, but which is non-fatal to have garbage collected if it has not been accessed recently, or if the owning package has been uninstalled.
+As compared to [Artifacts](@ref), these containers of data are mutable and should be treated as ephemeral; all usage of scratchspaces should assume that the data stored within them could be gone by the next time your code is run.
+In the current implementation, scratchspaces are removed during Pkg garbage collection if the space has not been accessed for a period of time (see the `spaces_cleanup_period` keyword argument to [Pkg.gc](@ref)), or if the owning package has been removed.
+Users can also request a full wipe of all spaces to clean up unused disk space through `Pkg.Spaces.clear_spaces!()`.
+
+## API overview
+
+Scratchspace usage is performed primarily through one function: `get_space!()`.
+It provides a single interface for creating and getting previously-created spaces, either tied to a package by its UUID, or as a global scratch space that can be accessed by any package.
+Here is an example where a package creates a space that is namespaced to its own UUID:
+
+```julia
+module SpaceExample
+using Pkg, Pkg.Spaces
+
+# This will be filled in inside `__init__()`
+download_cache = ""
+
+# Downloads a resource, stores it within a scratchspace
+function download_dataset(url)
+    fname = joinpath(download_cache, basename(url))
+    if !isfile(fname)
+        download(url, fname)
+    end
+    return fname
+end
+
+function __init__()
+    global download_cache = @get_space!("downloaded_files")
+end
+
+end # module SpaceExample
+```
+
+Note that we initialize the `download_cache` within `__init__()` so that our packages are as relocatable as possible; we typically do not want to bake absolute paths into our precompiled files.
+This makes use of the `@get_space!()` macro, which is identical to the `get_space!()` method, except it automatically determines the UUID of the calling module, if possible.
+An equivalent (but more verbose) invocation is given here:
+```julia
+function __init__()
+    global download_cache = get_space!("downloaded_files"; pkg_uuid=Base.PkgId(@__MODULE__).uuid)
```

Comment:
This feels quite a bit too internally to have in the docs. Couldn't `get_space!` take either a `Module` or a `UUID` and then dispatch accordingly?

Reply:
I'm confused; this is `get_space!()` taking in a `UUID` and dispatching accordingly?  We could of course have a method that takes in a `Module` as well.  Is that what you're suggesting?

Reply:
ah, you mean literally take either a `Module` or `UUID`, I understand now.  I mis-parsed your sentence.  :)